src/main/java/Ui.java

```diff
@@ -0,0 +1,223 @@
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+/**
+ * The UI of the application.
+ */
+public class Ui extends AnchorPane {
+
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Duke duke;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    /**
+     * Initializes the scroll pane to bind to the dialog container. Also prints out the welcome message on the chat.
+     */
+    @FXML
+    public void initialize() {
+
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        printGreeting();
+    }
+
+    /**
+     * Initializes duke.
+     *
+     * @param d the duke
+     */
+    public void setDuke(Duke d) {
+
+        duke = d;
+    }
+
+    /**
+     * Obtains the input command from the text field and creates a user dialog box containing the command. Clears the
+     * text field after creating the dialog box. The command is passed to Duke and processed to get the appropriate
+     * response, which is used to create another dialog box and appended to the dialog container.
+     */
+    @FXML
+    private void handleUserInput() {
+
+        String command = userInput.getText();
+
+        dialogContainer.getChildren().addAll(DialogBox.getUserDialog(command, userImage));
+        userInput.clear();
+
+        printResponse(duke.executeCommand(command));
+    }
+
+    /**
+     * Prints the welcoming message in the chat.
+     */
+    public void printGreeting() {
+
+        String greeting = "Hi! I'm Duke.\nHow can I help you?";
+
+        dialogContainer.getChildren().addAll(DialogBox.getDukeDialog(greeting, dukeImage));
+    }
+
+    /**
+     * Creates a dialog box containing the relevant response and append it onto the dialog container.
+     * Will call for exit if bye is received.
+     *
+     * @param response the response.
+     */
+    private void printResponse(Response response) {
+
+        String message = "";
+        switch (response.getMessage()) {
+        case ADD_TASK:
+            message = "I've got your back. Adding the new task:\n";
```

Comment:
Same here. If the messages are not going to change, then it is better to declare them as constant values. Even the constant strings can be formatted dynamically with different values.